clippy_lints/src/iter_without_into_iter.rs

```diff
@@ -56,7 +102,77 @@ fn is_nameable_in_impl_trait(ty: &rustc_hir::Ty<'_>) -> bool {
     !matches!(ty.kind, TyKind::OpaqueDef(..))
 }
 
+fn type_has_inherent_method(cx: &LateContext<'_>, ty: Ty<'_>, method_name: Symbol) -> bool {
+    if let Some(ty_did) = ty.ty_adt_def().map(ty::AdtDef::did) {
+        cx.tcx.inherent_impls(ty_did).iter().any(|did| {
+            cx.tcx
+                .hir()
+                .expect_item(did.expect_local())
+                .expect_impl()
+                .items
+                .iter()
+                .any(|item| item.ident.name == method_name)
```

Comment:
This would be slightly better using the `associated_items` query rather than going through the HIR tree.